### PR TITLE
Implemented akinci scheme. Example appended.

### PR DESCRIPTION
**Pull request by dineshadepu**
Implemented Akinci scheme correctly. 

I appended an example to check it. Please run the example with different density values i.e.,

change solid_rho to different values, and check it.

### CODE REVIEW

File: pysph/examples/rigid_body/sphere_in_vessel_akinci.py [hunk 1]
```diff
@@ -0,0 +1,220 @@
...
+    _tmp = gamma / (rho_0 * c_0**2)
+
+    rho = np.zeros_like(y)
+    for i in range(len(rho)):
+        p_i = rho_0 * 9.81 * (height_water_clmn - y[i])
+        rho[i] = rho_0 * (1 + p_i * _tmp)**(1. / gamma)
+    return rho
+
+
+def geometry():
```
Comment (prabhuramachandran) [inline pysph/examples/rigid_body/sphere_in_vessel_akinci.py]: Please use the functions provided already in the pysph.tools.geometry module instead of these.  This plotting function is also not needed here.

File: pysph/examples/rigid_body/sphere_in_vessel_akinci.py [hunk 2]
```diff
@@ -0,0 +1,220 @@
+"""A sphere of density 500 falling into a hydrostatic tank (15 minutes)
+
+Check basic equations of SPH to throw a ball inside the vessel
+"""
+from __future__ import print_function
+import numpy as np
+import matplotlib.pyplot as plt
```
Comment (prabhuramachandran) [inline pysph/examples/rigid_body/sphere_in_vessel_akinci.py]: Do not do this as our test suite is sometimes run on machines without matplotlib and this import may fail.  No need for this.

File: pysph/examples/rigid_body/sphere_in_vessel_akinci.py [hunk 3]
```diff
@@ -0,0 +1,220 @@
...
+from pysph.sph.wc.basic import TaitEOSHGCorrection, MomentumEquation
+from pysph.solver.application import Application
+from pysph.sph.rigid_body import (
+    BodyForce, SummationDensityBoundary, CutoffDensity, RigidBodyCollision,
+    LiuFluidForce, RigidBodyMoments, RigidBodyMotion, AkinciRigidFluidCoupling,
+    RK2StepRigidBody)
+
+
+def create_boundary():
+    dx = 2
```
Comment (prabhuramachandran) [inline pysph/examples/rigid_body/sphere_in_vessel_akinci.py]: These can all use pysph.tools.geometry.

File: pysph/examples/rigid_body/sphere_in_vessel_akinci.py [hunk 4]
```diff
@@ -0,0 +1,220 @@
...
+        V = np.ones_like(xt) * dx * dx * 1e-6
+        tank = get_particle_array_wcsph(x=xt, y=yt, h=h, m=m, rho=rho,
+                                        rad_s=rad_s, V=V, name="tank")
+        for name in ['fx', 'fy', 'fz']:
+            tank.add_property(name)
+
+        dx = 1
+        xc, yc = create_sphere(1)
+        m = np.ones_like(xc) * self.solid_rho * dx * 1e-3 * dx * 1e-3
+        rho = np.ones_like(xc) * self.solid_rho
```
Comment (prabhuramachandran) [inline pysph/examples/rigid_body/sphere_in_vessel_akinci.py]: With latest pysph you do not need ones_like for simple constants, just pass in a constant for the constant properties and it should work.  It will broadcast this to the same shape as x, y, z and use it.

File: pysph/sph/rigid_body.py [hunk 5]
```diff
@@ -225,13 +225,41 @@ def initialize(self, d_idx, d_m, d_fx, d_fy, d_fz):
...
+class CutoffDensity(Equation):
+    """Limit density to real fluid density
+
+
+    """
+    def __init__(self, dest, sources, fluid_rho=1000.0):
+        self.fluid_rho = fluid_rho
+        super(CutoffDensity, self).__init__(dest, sources)
+
+    def initialize(self, d_idx, d_rho):
```
Comment (prabhuramachandran) [inline pysph/sph/rigid_body.py]: Is this really needed?  Instead of changing the density directly, you should probably have a different equation of state that does not generate negative pressures by finding `max(rho0, d_rho[d_idx))` when computing the pressure.  Otherwise your `m_j/rho_j` terms will all be wrong.  So in short, don't change the rho but use a different EOS.

File: pysph/sph/rigid_body.py [hunk 6]
```diff
@@ -326,6 +394,7 @@ def __init__(self, dest, sources):
 
     def loop(self, d_idx, d_m, d_rho, d_au, d_av, d_aw,  d_p,
              s_idx, s_V, s_fx, s_fy, s_fz, DWIJ, s_m, s_p, s_rho):
+
```
Comment (prabhuramachandran) [inline pysph/sph/rigid_body.py]: Please don't add unnecessary whitespace.

**Comment by dineshadepu** [comment]
Merge this for now If it is urgent. Or else we can close this and merge when 3d example is ready.

**Comment by dineshadepu** [comment]
Closing this for now. Reopen it once 3d is done.
